JSDoc `@private`/`@protected` on a constructor is ignored

## Steps to reproduce

`main.js`:
```js
class C {
  /** @private */
  constructor() {}
}
new C();
```

```sh
tsgo --checkJs main.js
```

## Behavior with `typescript@6.0`

```
main.js(5,1): error TS2673: Constructor of class 'C' is private and only accessible within the class declaration.
```

## Behavior with `tsgo`

No error.

--- a/internal/ast/ast.go
+++ b/internal/ast/ast.go
@@ -1522,7 +1522,8 @@ func (node *ExportableBase) ExportableData() *ExportableBase { return node }
 
 // ModifiersBase
 
-func (node *ModifiersBase) Modifiers() *ModifierList { return node.modifiers }
+func (node *ModifiersBase) Modifiers() *ModifierList             { return node.modifiers }
+func (node *ModifiersBase) setModifiers(modifiers *ModifierList) { node.modifiers = modifiers }
 
 // LocalsContainerBase
 

--- a/internal/parser/reparser.go
+++ b/internal/parser/reparser.go
@@ -470,7 +470,7 @@ func (p *Parser) reparseHosted(tag *ast.Node, parent *ast.Node, jsDoc *ast.Node)
 			parent = parent.Expression()
 		}
 		switch parent.Kind {
-		case ast.KindPropertyDeclaration, ast.KindMethodDeclaration, ast.KindGetAccessor, ast.KindSetAccessor, ast.KindBinaryExpression:
+		case ast.KindPropertyDeclaration, ast.KindMethodDeclaration, ast.KindConstructor, ast.KindGetAccessor, ast.KindSetAccessor, ast.KindBinaryExpression:
 			var keyword ast.Kind
 			switch tag.Kind {
 			case ast.KindJSDocReadonlyTag:

--- a/testdata/baselines/reference/compiler/jsDocPrivateConstructor.errors.txt
+++ b/testdata/baselines/reference/compiler/jsDocPrivateConstructor.errors.txt
@@ -1,0 +1,14 @@
+main.js(7,1): error TS2673: Constructor of class 'C' is private and only accessible within the class declaration.
+
+
+==== main.js (1 errors) ====
+    // https://github.com/microsoft/typescript-go/issues/4219
+    
+    class C {
+      /** @private */
+      constructor() {}
+    }
+    new C();
+    ~~~~~~~
+!!! error TS2673: Constructor of class 'C' is private and only accessible within the class declaration.
+    

--- a/testdata/baselines/reference/compiler/jsDocPrivateConstructor.symbols
+++ b/testdata/baselines/reference/compiler/jsDocPrivateConstructor.symbols
@@ -1,0 +1,14 @@
+//// [tests/cases/compiler/jsDocPrivateConstructor.ts] ////
+
+=== main.js ===
+// https://github.com/microsoft/typescript-go/issues/4219
+
+class C {
+>C : Symbol(C, Decl(main.js, 0, 0))
+
+  /** @private */
+  constructor() {}
+}
+new C();
+>C : Symbol(C, Decl(main.js, 0, 0))
+

--- a/testdata/baselines/reference/compiler/jsDocPrivateConstructor.types
+++ b/testdata/baselines/reference/compiler/jsDocPrivateConstructor.types
@@ -1,0 +1,15 @@
+//// [tests/cases/compiler/jsDocPrivateConstructor.ts] ////
+
+=== main.js ===
+// https://github.com/microsoft/typescript-go/issues/4219
+
+class C {
+>C : C
+
+  /** @private */
+  constructor() {}
+}
+new C();
+>new C() : C
+>C : typeof C
+

--- a/testdata/baselines/reference/submodule/conformance/jsdocReadonly.errors.txt
+++ b/testdata/baselines/reference/submodule/conformance/jsdocReadonly.errors.txt
@@ -1,7 +1,8 @@
+jsdocReadonly.js(13,9): error TS1024: 'readonly' modifier can only appear on a property declaration or index signature.
 jsdocReadonly.js(23,3): error TS2540: Cannot assign to 'y' because it is a read-only property.
 
 
-==== jsdocReadonly.js (1 errors) ====
+==== jsdocReadonly.js (2 errors) ====
     class LOL {
         /**
          * @readonly
@@ -15,6 +16,8 @@ jsdocReadonly.js(23,3): error TS2540: Cannot assign to 'y' because it is a read-
         /** @readonly Definitely not here */
         static z = 3
         /** @readonly This is OK too */
+            ~~~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS1024: 'readonly' modifier can only appear on a property declaration or index signature.
         constructor() {
             /** ok */
             this.y = 2

--- a/testdata/baselines/reference/submodule/conformance/jsdocReadonly.errors.txt.diff
+++ b/testdata/baselines/reference/submodule/conformance/jsdocReadonly.errors.txt.diff
@@ -1,0 +1,21 @@
+--- old.jsdocReadonly.errors.txt
++++ new.jsdocReadonly.errors.txt
+@@= skipped -0, +0 lines =@@
++jsdocReadonly.js(13,9): error TS1024: 'readonly' modifier can only appear on a property declaration or index signature.
+ jsdocReadonly.js(23,3): error TS2540: Cannot assign to 'y' because it is a read-only property.
+
+
+-==== jsdocReadonly.js (1 errors) ====
++==== jsdocReadonly.js (2 errors) ====
+     class LOL {
+         /**
+          * @readonly
+@@= skipped -14, +15 lines =@@
+         /** @readonly Definitely not here */
+         static z = 3
+         /** @readonly This is OK too */
++            ~~~~~~~~~~~~~~~~~~~~~~~~~
++!!! error TS1024: 'readonly' modifier can only appear on a property declaration or index signature.
+         constructor() {
+             /** ok */
+             this.y = 2

--- a/testdata/tests/cases/compiler/jsDocPrivateConstructor.ts
+++ b/testdata/tests/cases/compiler/jsDocPrivateConstructor.ts
@@ -1,0 +1,11 @@
+// @checkJs: true
+// @noEmit: true
+// @filename: main.js
+
+// https://github.com/microsoft/typescript-go/issues/4219
+
+class C {
+  /** @private */
+  constructor() {}
+}
+new C();

